packages/fhir-converter/src/lib/handlebars-converter/handlebars-helpers.js

```diff
@@ -601,17 +602,18 @@ module.exports.external = [
     func: function getFirstCdaSectionsByTemplateId(msg, ...templateIds) {
       try {
         var ret = {};
-
-        for (var t = 0; t < templateIds.length - 1; t++) {
-          //-1 because templateIds includes the full message at the end
+  
+        for (var t = 0; t < templateIds.length - 1; t++) { // -1 because templateIds includes the full message at the end
           for (var i = 0; i < msg.ClinicalDocument.component.structuredBody.component.length; i++) {
             let sectionObj = msg.ClinicalDocument.component.structuredBody.component[i].section;
-            if (
-              sectionObj.templateId &&
-              JSON.stringify(sectionObj.templateId).includes(templateIds[t])
-            ) {
+            // Normalize sectionObj.templateId to always be an array
+            let templateIdsArray = Array.isArray(sectionObj.templateId) ? sectionObj.templateId : [sectionObj.templateId];
+            
+            // Check if any templateId object in the normalized array has a root value that matches exactly
+            const hasExactMatch = templateIdsArray.some(templateIdObj => templateIdObj && templateIdObj.root === templateIds[t]);
```

Comment:
exact match won't cause us to miss sections that relied on this?

not sure if there's any, but I assume this was like this for a reason

Reply:
So it didnt matter before because they never used the optional ids, which are shorter than the required ones. For immunizations
2.16.840.1.113883.10.20.22.2.2   -> optional
2.16.840.1.113883.10.20.22.2.2.1 > required

they didnt have the optional one before, and the full one didnt overlap with any other resources. But the optional one if you use include matches with:

2.16.840.1.113883.10.20.22.2.23 -> MedicationEquipment

which means if we use include we lost immunisations if the first resource we find is a MedicationEquipment.  

Tldr: is that we need exact match since we are using the shorter optional Ids, but we arent missing any data, which makes sense logically and also is shown emprically in test counts